app/src/main/java/com/springboot/webpush/controller/VAPIDController.java

```diff
@@ -1,27 +1,36 @@
 package com.springboot.webpush.controller;
 
-import java.lang.invoke.MethodHandles;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 import com.springboot.webpush.service.VAPIDService;
 
 @RestController
 public class VAPIDController {
-	private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
-	private final VAPIDService vapidService;
-
-	public VAPIDController(final VAPIDService vapidService) {
-		super();
-		this.vapidService = vapidService;
-	}
-
-	@GetMapping("/api/vapid-public")
-	public String vapidPublic() {
-		return vapidService.getPublicKey();
-	}
+    private final VAPIDService vapidService;
+
+    public VAPIDController(final VAPIDService vapidService) {
+        super();
+        this.vapidService = vapidService;
+    }
+
+    @GetMapping("/api/vapid-public")
+    ResponseEntity<String> getVapidPublic() {
+
+        final var keyStore = vapidService.getKeyStore();
+        if (keyStore.valid()) {
+            return ResponseEntity.ok(keyStore.getPublicKey());
+        }
+        return ResponseEntity.notFound().build();
+    }
+
+    @PostMapping("/api/vapid")
+    ResponseEntity<String> regenerate() {
```

Comment:
this is probably fine for our prototype, but I don't think we want to actually expose this functionality in the API, do we? What would be the use case?

Reply:
I agree. I converted this function to a command line utility